internal/dataplane/parser/translate_grpcroute.go

```diff
@@ -99,10 +99,55 @@ func generateKongRoutesFromGRPCRouteRule(grpcroute *gatewayv1alpha2.GRPCRoute, r
 			},
 		}
 
+		// Kong routes derived from a GRPCRoute use a path composed of the match's gRPC service and method
+		// If either the service or method is omitted, there is a default regex determined by the match type
+		// https://gateway-api.sigs.k8s.io/geps/gep-1016/#matcher-types describes the defaults
+		if match.Method != nil {
+			var method, service string
+			matchMethod := match.Method.Method
+			matchService := match.Method.Service
+			var matchType gatewayv1alpha2.GRPCMethodMatchType
+			if match.Method.Type == nil {
+				matchType = gatewayv1alpha2.GRPCMethodMatchExact
+			} else {
+				matchType = *match.Method.Type
+			}
+			serviceMap := map[gatewayv1alpha2.GRPCMethodMatchType]string{
+				gatewayv1alpha2.GRPCMethodMatchType(""):          ".+",
+				gatewayv1alpha2.GRPCMethodMatchExact:             ".+",
+				gatewayv1alpha2.GRPCMethodMatchRegularExpression: ".+",
+			}
+			methodMap := map[gatewayv1alpha2.GRPCMethodMatchType]string{
+				gatewayv1alpha2.GRPCMethodMatchType(""):          "",
+				gatewayv1alpha2.GRPCMethodMatchExact:             "",
+				gatewayv1alpha2.GRPCMethodMatchRegularExpression: ".+",
+			}
+			if matchMethod == nil {
+				method = methodMap[matchType]
+			} else {
+				method = *matchMethod
+			}
+			if matchService == nil {
+				service = serviceMap[matchType]
+			} else {
+				service = *matchService
+			}
+			r.Paths = append(r.Paths, kong.String(fmt.Sprintf("~/%s/%s", service, method)))
+		}
+
 		if len(grpcroute.Spec.Hostnames) > 0 {
 			r.Hosts = getGRPCRouteHostnamesAsSliceOfStringPointers(grpcroute)
 		}
 
+		r.Headers = map[string][]string{}
+		for _, hmatch := range match.Headers {
+			name := string(hmatch.Name)
+			if _, ok := r.Headers[name]; !ok {
```

Comment:
There is no need to do the check, just do the `append` directly without checking whether the value is present.